src/planwise/client/datasets/components/listing.cljs

```diff
@@ -0,0 +1,74 @@
+(ns planwise.client.datasets.components.listing
+  (:require [re-frame.core :refer [subscribe dispatch]]
+            [planwise.client.utils :as utils]
+            [planwise.client.components.common :as common]
+            [planwise.client.components.progress-bar :as progress-bar]
+            [planwise.client.datasets.db :as db]))
+
+
+(defn new-dataset-button
+  []
+  [:button.primary
+   {:on-click
+    #(dispatch [:datasets/begin-new-dataset])}
+   "New Dataset"])
+
+(defn search-box
+  [dataset-count]
+  (let [search-string (subscribe [:datasets/search-string])]
+    (fn [dataset-count]
+      [:div.search-box
+       [:div (utils/pluralize dataset-count "dataset")]
+       [:input
+        {:type "search"
+         :placeholder "Search datasets..."
+         :value @search-string
+         :on-change #(dispatch [:datasets/search (-> % .-target .-value str)])}]
+       [new-dataset-button]])))
+
+(defn no-datasets-view []
+  [:div.empty-list
+   [:img {:src "/images/empty-datasets.png"}]
+   [:p "You have no datasets yet"]
+   [:div
+    [new-dataset-button]]])
+
+(defn dataset-card
+  [{:keys [id name description facility-count project-count server-status] :as dataset}]
+  (let [state (db/server-status->state server-status)
+        importing? (db/dataset-importing? state)
+        cancelling? (db/dataset-cancelling? state)]
+    [:div.dataset-card
+     [:h1 name]
+     [:h2 description]
+     [:p
+      [common/icon :location "icon-small"]
+      (utils/pluralize facility-count "facility" "facilities")
+      (str " (" (db/server-status->string server-status) ")")]
+     (when importing?
+       [progress-bar/progress-bar (db/import-progress server-status)])
+     (cond
+       importing?
+       [:div.bottom-right
+        [:button.danger
+         {:type :button
+          :on-click #(dispatch [:datasets/cancel-import! id])
+          :disabled cancelling?}
+         (if cancelling? "Cancelling..." "Cancel")]]
+
+       (zero? project-count)
+       [:div.bottom-right
+        [:button.delete
+         {:on-click (utils/with-confirm
+                      #(dispatch [:datasets/delete-dataset id])
+                      "Are you sure you want to delete this dataset?")}
+         "\u2716 Delete"]])]))
```

Comment:
We might want to add a "Cannot delete as there are projects associated" label, to explain to the user why a dataset can be deleted while another one is not offering that option